### PR TITLE
jwt implementation

### PR DESCRIPTION
**Pull request by ADIGrimm**

### CODE REVIEW

File: src/main/java/online/bookstore/controller/AuthenticationController.java [hunk 1]
```diff
@@ -15,10 +18,16 @@
...
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto createBook(@Valid @RequestBody UserRegistrationRequestDto requestDto) {
         return userService.registration(requestDto);
     }
 
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/controller/AuthenticationController.java]:
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
```

File: src/main/java/online/bookstore/controller/AuthenticationController.java [hunk 2]
```diff
@@ -15,10 +18,16 @@
...
 public class AuthenticationController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto createBook(@Valid @RequestBody UserRegistrationRequestDto requestDto) {
         return userService.registration(requestDto);
     }
 
+    @PostMapping("/login")
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/controller/AuthenticationController.java:31]: Add Swagger annotations

File: src/main/java/online/bookstore/dto/user/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package online.bookstore.dto.user;
+
+public record UserLoginRequestDto(String email, String password) {
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/dto/user/UserLoginRequestDto.java]: add validation

File: src/main/java/online/bookstore/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/online/bookstore/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(constant.length());
```

File: src/main/java/online/bookstore/security/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/security/JwtAuthenticationFilter.java]: create constant for `"Bearer "`

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -6,3 +6,6 @@ spring.datasource.password=
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+
+jwt.expiration=300000
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties:10]: don't forget to add these properties to test/app.properties

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/online/bookstore/security/JwtAuthenticationFilter.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    public static final String TOKEN_TYPE = "Bearer ";
```
Comment (liliia-ponomarenko) [inline src/main/java/online/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
    private static final String TOKEN_TYPE = "Bearer ";
```

**Review by liliia-ponomarenko — APPROVED** [review]
Great job!

**Pull request merged**